### ISSUE TITLE
Collect compiler warnings on SonarQube

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#1396]
The compiler warnings are important and deserve to be tracked in SonarQube, especially since we generally try to write analyzers that do not raise the same messages as the C# compiler. In some cases it could be very important to track compiler warnings - for example, the following code will generate a compiler warning, but not a SonarC# issue and is still important enough to be tracked and fixed:
```
public void Foo(int a)
{
	return;
	this.a = a;
}
```

**Comment by Evangelink** [comment]
Closing this ticket but maybe we should do something as part of the be friendly with third-party analyzers.
